pika/connection.py

```diff
@@ -157,6 +157,32 @@ def __repr__(self):
                 (self.__class__.__name__, self.host, self.port,
                  self.virtual_host, bool(self.ssl_options)))
 
+    def __eq__(self, other):
+        if isinstance(self, other.__class__):
+            return (
+                self._backpressure_detection == other._backpressure_detection
```

Comment:
RabbitMQ identifies connections using a 4-tuple: client hostname, client port, server hostname, server port. Should the same logic be applied here? Of course it would assume that the connection has an open socket.

Reply:
Do you mean we should only check the equality of these 4 parameters instead of all the parameters?

Reply:
It makes sense to use the same comparison logic for connections that RabbitMQ does.

Reply:
Alright. I’m going to correct this and make some tests.

Reply:
@michaelklishin @lukebakken Sorry I misread. I am not sure I understand now. Which connection parameters do you suggest we compare exactly? You are talking about *client* and *server* hostnames and ports, but as far as I know we only have access to the server hostnames and ports (`self._host` and `self._port`).

Reply:
Oh, right. Hmm it would be nice to have access to the ephemeral ports in use as that would identify the connection. That seems like the correct way to compare two connections.